packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/EditingContextCrossReferenceAdapter.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2022 CEA.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.emf.services;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import org.eclipse.emf.common.notify.Notification;
+import org.eclipse.emf.common.util.TreeIterator;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.EStructuralFeature.Setting;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.util.ECrossReferenceAdapter;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+
+/**
+ * A {@link ECrossReferenceAdapter}@ that is able to clean dangling references when an object is deleted or when a
+ * resource is removed from the ResourceSet.
+ *
+ * @author lfasani
+ */
+public class EditingContextCrossReferenceAdapter extends ECrossReferenceAdapter {
+
+    @Override
+    protected void handleContainment(Notification notification) {
+        Object oldValue = notification.getOldValue();
+        int eventType = notification.getEventType();
+
+        if (eventType == Notification.REMOVE || eventType == Notification.REMOVE_MANY) {
```

Comment:
The enclosing `if (eventType == Notification.REMOVE || eventType == Notification.REMOVE_MANY)` seems unneeded.

Reply:
Why? I do not want to do unnecessary work

Reply:
This would be simpler and not perform any test twice:

```java
if (eventType == Notification.REMOVE) {
  this.handleRemoveObject(oldValue);
} else if (eventType == Notification.REMOVE_MANY) {
  for (Object object : (Collection<?>) oldValue) {
    this.handleRemoveObject(object);
  }
} else {
  super.handleContainment(notification);
}
```